### PR TITLE
IDEM-303

### PR DESCRIPTION
**Pull request by JasminFarhi**
#303 
Add validation to email claim
IMG_7879

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,9 +33,11 @@
     "typescript": "^4.2.3"
   },
   "dependencies": {
+    "install": "^0.13.0",
     "moment": "^2.29.4",
     "nodejs-nodemailer-outlook": "^1.2.4",
     "react-native-modal": "^13.0.1",
-    "ts-node": "^10.8.1"
+    "ts-node": "^10.8.1",
+    "validator": "^13.7.0"
```
Comment (xiangzhe0812) [inline package.json:41]: This guy should be in app/package.json

**Review by bitcoinbrisbane — DISMISSED** [review]

**Comment by bitcoinbrisbane** [comment]
Might need to fix this 
image

File: app/src/navigation/screens/ClaimScreen.tsx [hunk 2]
```diff
@@ -107,16 +108,22 @@ const ClaimScreen: React.FC = () => {
...
-    await addClaim(claim.type, formState, selectedFileIds);
-
-    if (claim.type === "EmailCredential") {
-      const email = (formState.email as string).toLowerCase();
-      await verifyPublicKey(email);
+    if (
+      claim.type === "EmailCredential" &&
+      !isEmail(formState.email as string)
+    ) {
+      Alert.alert(
```
Comment (xiangzhe0812) [inline app/src/navigation/screens/ClaimScreen.tsx:119]: Need return here, we dont want to save the email not valid.

Comment (JasminFarhi) [reply app/src/navigation/screens/ClaimScreen.tsx:118]: added a return :)

**Review by bitcoinbrisbane — APPROVED** [review]

File: app/src/navigation/screens/ClaimScreen.tsx [hunk 3]
```diff
@@ -107,16 +108,22 @@ const ClaimScreen: React.FC = () => {
...
 
   const onSave = async () => {
     setLoading(true);
-    await addClaim(claim.type, formState, selectedFileIds);
-
-    if (claim.type === "EmailCredential") {
-      const email = (formState.email as string).toLowerCase();
-      await verifyPublicKey(email);
+    if (
+      claim.type === "EmailCredential" &&
```
Comment (amagon270) [inline app/src/navigation/screens/ClaimScreen.tsx:116]:
```suggestion
      isEmailClaim &&
```

**Review by amagon270 — CHANGES_REQUESTED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by amagon270 — APPROVED** [review]

**Pull request merged**